charts/redpanda/templates/tests/test-loadbalancer-tls.yaml

```diff
@@ -0,0 +1,127 @@
+{{/*
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+  
+  http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  */}}
+{{- if and .Values.tls.enabled ( eq .Values.external.type "LoadBalancer" ) -}}
+  {{- $values := .Values }}
+  {{- $root := deepCopy . }}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ include "redpanda.fullname" . }}-test-loadbalancer-tls
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+  {{- with include "full.labels" . }}
+  {{- . | nindent 4 }}
+  {{- end }}
+  annotations:
+    "helm.sh/hook": test
+    "helm.sh/hook-delete-policy": before-hook-creation
+spec:
+  serviceAccountName: default
+  restartPolicy: Never
+  securityContext:
+    runAsUser: 65535
+    runAsGroup: 65535
+  {{- with .Values.imagePullSecrets }}
+  imagePullSecrets: {{- toYaml . | nindent 4 }}
+  {{- end }}
+  containers:
+    - name: {{ template "redpanda.name" . }}
+      image: {{ .Values.image.repository }}:{{ template "redpanda.tag" . }}
+      command:
+        - /usr/bin/timeout
+        - "120"
+        - bash
+        - -c
+        - |
+          set -xe
+
+  {{- range $name, $cert := $values.tls.certs }}
+  {{- if $cert.secretRef }}
+    {{- if eq $cert.secretRef.name "external-tls-secret" }}
+          echo "---> testing external tls"
+  
+          # in the future remove this requirement
+          ip_list="172.19.200.200 172.19.200.201 172.19.200.202"
```

Comment:
Shouldn't we pass this as helm chart values? If user would like to run helm test when loadblanacer is enabled, then test would fail right?

Reply:
Very good point, one I have not really considered since tests also assume certain assets exist  that do not necessarily exist for user deployments. Those will also fail right?

Reply:
I updated to get the list dynamically, but I cannot guarantee it will always work in the wild.